src/main/java/com/sourcegraph/cody/agent/CodyAgentCodebase.kt

```diff
@@ -4,46 +4,46 @@ import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VirtualFile
 import com.sourcegraph.cody.CodyToolWindowContent
+import com.sourcegraph.cody.config.CodyProjectSettings
 import com.sourcegraph.config.ConfigUtil
 import com.sourcegraph.vcs.RepoUtil
 
-class CodyAgentCodebase(private val underlying: CodyAgentServer, val project: Project) {
+class CodyAgentCodebase(private val underlying: CodyAgentServer, private val project: Project) {
 
-  // TODO: This should ideally be persisted as a project-wide setting. Down the road, it should also
-  // support a list of repository names instead of only a single codebase.
-  private var inferredCodebase: String = ""
-  private var explicitCodebase: String = ""
+  // TODO: Support list of repository names instead of just one.
+  private val application = ApplicationManager.getApplication()
+  private val settings = CodyProjectSettings.getInstance(project)
 
   init {
-    ApplicationManager.getApplication().executeOnPooledThread {
-      onRepositoryName(RepoUtil.findRepositoryName(project, null))
+    application.executeOnPooledThread {
+      onRepositoryName(settings.remoteUrl ?: RepoUtil.findRepositoryName(project, null))
     }
   }
 
-  fun onNewExplicitCodebase(codebase: String) {
-    explicitCodebase = codebase
+  fun setUrl(url: String) {
+    settings.remoteUrl = url
     onPropagateConfiguration()
   }
 
-  fun currentCodebase(): String? = explicitCodebase.ifEmpty { inferredCodebase }.ifEmpty { null }
+  fun getUrl(): String? = settings.remoteUrl
 
   fun onFileOpened(project: Project, file: VirtualFile) {
-    ApplicationManager.getApplication().executeOnPooledThread {
-      onRepositoryName(RepoUtil.findRepositoryName(project, file))
+    application.executeOnPooledThread {
+      onRepositoryName(settings.remoteUrl ?: RepoUtil.findRepositoryName(project, file))
     }
   }
 
-  private fun onPropagateConfiguration() {
-    CodyToolWindowContent.getInstance(project).embeddingStatusView.updateEmbeddingStatus()
-    underlying.configurationDidChange(ConfigUtil.getAgentConfiguration(project))
-  }
-
-  private fun onRepositoryName(repositoryName: String?) {
-    ApplicationManager.getApplication().invokeLater {
-      if (repositoryName != null && inferredCodebase != repositoryName) {
-        inferredCodebase = repositoryName
+  private fun onRepositoryName(url: String?) {
```

Comment:
I do not understand this method. Its name is unclear to me. Can you rename it?

Reply:
It's the same name as before this PR. A clearer name might have been `onRepositoryNameChanged` or something like that